components/UI/searchResult.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { FunctionComponent, useEffect, useState } from "react";
+import styles from "../../styles/search.module.css";
+import SearchBadge from "./searchBadge";
+
+type SearchResultProps = {
+  search: (result: SearchResult) => void;
+  currentResult: SearchResult | null | undefined;
+  history: SearchResult[];
+  showHistory: boolean;
+};
+
+const SearchResult: FunctionComponent<SearchResultProps> = ({
+  search,
+  currentResult,
+  history,
+  showHistory,
+}) => {
+  const [searchResults, setSearchResults] = useState<SearchResult[]>();
+  useEffect(() => {
+    if (!history) return;
+    if (!currentResult || currentResult.name === "") {
+      setSearchResults(history);
+      return;
+    }
+    const filtered = history.filter(
+      (result) => result.name !== currentResult?.name
+    );
+    setSearchResults(
+      filtered.length > 4 && currentResult?.name !== ""
+        ? filtered.slice(0, 4)
+        : filtered
+    );
+  }, [currentResult]);
```

Comment:
history is missing from the dependencies of the useEffect, I think you should add it to avoid bugs